src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/InterventionController.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+import org.springframework.web.server.ServerWebInputException
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.InterventionDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.service.InterventionService
+import java.util.UUID
+
+@RestController
+class InterventionController(
+  private val interventionService: InterventionService
+) {
+
+  @GetMapping("/intervention/{id}")
+  fun getInterventionByID(@PathVariable id: String): InterventionDTO {
+    val uuid = parseID(id)
+
+    return interventionService.getIntervention(uuid)
+      ?.let { InterventionDTO.from(it) }
+      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "intervention not found [id=$uuid]")
+  }
+
+  private fun parseID(id: String): UUID {
```

Comment:
this is repeated from elsewhere. let's refactor to avoid this.

Reply:
Pulled into separate StringHelper file.